ff/src/biginteger/mod.rs

```diff
@@ -870,6 +872,326 @@ impl<const N: usize> Not for BigInt<N> {
     }
 }
 
+/// Computes a division between to BigInt with remainder.
+///
+/// This algorithm is taken from "The Art of Computer Programming" by Donald
+/// Knuth, Third Edition, Section 4.3.2, Algorithm D, with some elements taken
+/// from the repository https://github.com/rust-num/num-bigint.
+fn div_with_rem<B: Borrow<BigInt<N>>, const N: usize>(
+    a: BigInt<N>,
+    d: B,
+) -> (BigInt<N>, BigInt<N>) {
+    // Basic cases
+    if d.borrow().is_zero() {
+        panic!("trying to divide by zero");
+    }
+    if d.borrow().is_one() {
+        return (a, BigInt::<N>::zero());
+    }
+    if a.is_zero() {
+        return (BigInt::<N>::zero(), BigInt::<N>::zero());
+    }
+
+    // Covers the case of N = 1, for which the quotient and remainder are
+    // computed as usual.
+    if N == 1 {
+        let q: u64 = a.0[0] / d.borrow().0[0];
+        let r: u64 = a.0[0] % d.borrow().0[0];
+        return (BigInt::<N>::from(q), BigInt::<N>::from(r));
+    }
+
+    // Covers immediate cases where a <= d.
+    match a.cmp(d.borrow()) {
```

Comment:
I think for simplicity we can add a line that borrows `d` just once:

```suggestion
	let d = d.borrow();

    // Covers immediate cases where a <= d.
    match a.cmp(d) {
```